Shared/Tests.Shared/NotificationTests.cs

```diff
@@ -252,6 +275,143 @@ public void CollectionChanged_WhenTransactionHasBothAddAndRemove_ShouldReset()
             }
         }
 
+        [Test]
+        public void List_WhenUnsubscribed_ShouldStopReceivingNotifications()
+        {
+            var container = new OrderedContainer();
+            _realm.Write(() => _realm.Add(container));
+
+            var eventArgs = new List<NotifyCollectionChangedEventArgs>();
+            var handler = new NotifyCollectionChangedEventHandler((sender, e) =>
+            {
+                eventArgs.Add(e);
+            });
+
+            var collection = container.Items.AsRealmCollection();
+            collection.CollectionChanged += handler;
+
+            _realm.Write(() =>
+            {
+                container.Items.Add(new OrderedObject());
+            });
+
+            TestHelpers.RunEventLoop();
+
+            Assert.That(eventArgs.Count, Is.EqualTo(1));
+            Assert.That(eventArgs[0].Action, Is.EqualTo(NotifyCollectionChangedAction.Add));
+
+            collection.CollectionChanged -= handler;
+
+            _realm.Write(() =>
+            {
+                container.Items.Add(new OrderedObject());
+            });
+
+            TestHelpers.RunEventLoop();
+
+            Assert.That(eventArgs.Count, Is.EqualTo(1));
+            Assert.That(eventArgs[0].Action, Is.EqualTo(NotifyCollectionChangedAction.Add));
+        }
+
+        [Test]
+        public void List_WhenTransactionHasBothAddAndRemove_ShouldReset()
```

Comment:
This test bothers me as either I don't understand the semantics or they are wrong. It seems to say that if you add an item you still get **only** a reset event, presumably from the Clear. Assuming all the code is correct,  could you add a couple of comments to clarify what bits are setup and what causes the events being asserted.